portfolio/src/main/java/com/google/sps/PuzzleDao.java

```diff
@@ -85,4 +92,35 @@ public void delete(Long puzzleId) {
       //deletes the entity whose id match "puzzleID"
       datastore.delete(keyFactory.newKey(puzzleId));
     }
+
+    public ListPuzzles listPuzzles(String cursorUrl, int PAGE_SIZE) {
```

Comment:
nit: make this `pageSize`. Generally only static constant variables are written in underscore case.